lsst_efd_client/efd_helper.py

```diff
@@ -98,8 +98,10 @@ def build_time_range_query(self, topic_name, fields, start, end, is_window=False
         else:
             raise TypeError('The second time argument must be the time stamp for the end ' +
                             'or a time delta.')
-
-        timespan = f"time >= '{start_str}Z' AND time <= '{end_str}Z'"  # influxdb demands the trailing Z
+        index_str = ''
+        if index:
+            index_str = f' AND {topic_name}ID = {index}'
```

Comment:
This isn't going to work. You need to parse out the CSC name. Looking at your example notebooks, if you use this topic, _lsst.sal.ATCamera.wreb_, you'll get _lsst.sal.ATCamera.wrebID_ which isn't the expected _ATCameraID_ field.

Reply:
You are absolutely right.  I was too hasty creating this pull request.  Sorry about that.  I'll try to figure out the best way to parse out the CSC name.

Reply:
OK, I've now tested this with one indexed CSC and it seemed to do the right thing.  I also include another commit which prevents saving creds in the constructed object.  This isn't too risky, but if the object were pickled, someone could possibly retrieve them.

Reply:
Good catch on the credential caching.